conan/tools/gnu/mingw.py

```diff
@@ -0,0 +1,13 @@
+def is_mingw(conanfile):
+    """
+    Validate if current compiler in host setttings is related to MinGW
+    :param conanfile: ConanFile instance
+    :return: True, if the host compiler is related to MinGW, otherwise False.
+    """
+    host_os = conanfile.settings.get_safe("os")
```

Comment:
Also adding my comment here[[1]](https://github.com/conan-io/conan-center-index/pull/18283#discussion_r1250611884):

> This feels like a lot of "work" that will result in false in probably most cases (I don't have numbers but I assume mingw has quite a bit lower market share compared to gcc and msvc")
I guess it would still be better to get the short circuit behaviour ~compared to the old code~ where it returns false as soon as some condition is reached that makes it false.

Reply:
I agree with this, part this PR never got prioritized is because the logic reads too compact, and it also introduces new aspects that are not common practice in ConanCenter recipes, in which the majority just check ``Windows && gcc => MinGW``

Reply:
> and it also introduces new aspects that are not common practice in ConanCenter recipes, in which the majority just check Windows && gcc => MinGW

This might be a new aspect, but it's an important one. According to https://blog.conan.io/2022/10/13/Different-flavors-Clang-compiler-Windows.html the compiler.runtime setting is the official way to distinguish between MinGW clang and VS clang-cl. Some recipes in Conan Center already take this into account, but most do not. This is one of the main reasons why we are struggling a lot with introducing Conan at our company (we use MinGW clang for compatibility and performance reasons).

This PR could help a lot with easier integration and broader adoption of this Windows clang distinction through Conan Center.

Reply:
Hi @mhthies. There is an open ticket for this helper here: https://github.com/conan-io/conan/issues/15648, lets try to follow up there.